### PR TITLE
Remove device connection will always remove cloud connection

### PR DESCRIPTION
**Pull request by philipktlin**

### CODE REVIEW

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs [hunk 1]
```diff
@@ -151,6 +151,7 @@ void RegisterMqttModule(ContainerBuilder builder, (bool isEnabled, bool usePersi
             bool encryptTwinStore = this.configuration.GetValue("EncryptTwinStore", false);
             int configUpdateFrequencySecs = this.configuration.GetValue("ConfigRefreshFrequencySecs", 3600);
             TimeSpan configUpdateFrequency = TimeSpan.FromSeconds(configUpdateFrequencySecs);
+            bool closeCloudConnectionWhenCloseDeviceConnection = this.configuration.GetValue("closeCloudConnectionWhenCloseDeviceConnection", true);
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs:154]:
```suggestion
            bool closeCloudConnectionWhenCloseDeviceConnection = this.configuration.GetValue("CloseCloudConnectionOnDeviceDisconnect", true);
```

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs [hunk 2]
```diff
@@ -29,13 +29,15 @@ public class ConnectionManager : IConnectionManager
         readonly ICredentialsCache credentialsCache;
         readonly IIdentityProvider identityProvider;
         readonly IDeviceConnectivityManager connectivityManager;
+        readonly bool closeCloudConnectionWhenCloseDeviceConnection;
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs:32]: Test case?

Comment (philipktlin) [reply edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/ConnectionManager.cs:32]: Add one test for RemoveDeviceConnection method.

File: edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs [hunk 3]
```diff
@@ -151,6 +151,7 @@ void RegisterMqttModule(ContainerBuilder builder, (bool isEnabled, bool usePersi
             bool encryptTwinStore = this.configuration.GetValue("EncryptTwinStore", false);
             int configUpdateFrequencySecs = this.configuration.GetValue("ConfigRefreshFrequencySecs", 3600);
             TimeSpan configUpdateFrequency = TimeSpan.FromSeconds(configUpdateFrequencySecs);
+            bool closeCloudConnectionWhenCloseDeviceConnection = this.configuration.GetValue("CloseCloudConnectionWhenCloseDeviceConnection", true);
```
Comment (varunpuranik) [inline edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Service/DependencyManager.cs:154]: Looks like my suggestion got overwritten? 
```suggestion
            bool closeCloudConnectionWhenCloseDeviceConnection = this.configuration.GetValue("CloseCloudConnectionOnDeviceDisconnect", true);
```

**Pull request merged**